homeassistant/components/russound_rnet/media_player.py

```diff
@@ -91,6 +95,11 @@ def __init__(self, hass, russ, sources, zone_id, extra):
         # Each zone resets to 1-6 per controller
         self._zone_id = (zone_id - 1) % 6 + 1
 
+        # Stable unique_id so entity get registered once
+        self._attr_unique_id = (
+            f"{device_key}_{self._controller_id}_zone_{self._zone_id}"
```

Comment:
I think this is okay, except for the fact that the device_key consists of a name. Names are a bad unique id as it will create new entities when someone changes the name, which is a bad user experience.

Since I believe controller_id and zone_id are just 0-n values, it means that you will have collisions when you have multiple devices, which is probably why you added the name to this.

If that is indeed the case, we should either find a way to get a serial number or another unique thing to identify the device, otherwise we need to add a config flow so we can use the entry_id here